### PR TITLE
Add description on enum constants

### PR DESCRIPTION
**Pull request by clementdenis**
This PR allows to set a description on enum values in API method parameters, as allowed in the discovery format.
- The DiscoveryGenerator [already supported enum descriptions](https://github.com/cloudendpoints/endpoints-java/blob/master/endpoints-framework/src/main/java/com/google/api/server/spi/discovery/DiscoveryGenerator.java#L230).
- The OpenAPI format does not support descriptions on enum values ([yet](https://github.com/OAI/OpenAPI-Specification/issues/348)).

PS: we tried to implement description for enums in Swagger [here](https://github.com/AODocs/endpoints-java/tree/enum_description_in_swagger), as described in [Swagger documentation for enums](https://swagger.io/docs/specification/data-models/enums/), using the description field, but we preferred not to include it yet.

### CODE REVIEW

**Comment by kaldebert** [comment]
I'm ok with my commits being contributed to this project.

**Comment by tangiel** [comment]
Looks like merging the other PR puts this in conflict. Please fix.

**Comment by kaldebert** [comment]
@tangiel The merge includes the fix for the currently failing master.
The last commit covers the missing case for description of $ref fields.

File: endpoints-framework/src/main/java/com/google/api/server/spi/config/model/SchemaRepository.java [hunk 1]
```diff
@@ -151,11 +152,14 @@ private Schema getOrCreateTypeForConfig(
       return MAP_SCHEMA;
     } else if (Types.isEnumType(type)) {
       Schema.Builder builder = Schema.builder()
-          .setName(Types.getSimpleName(type, config.getSerializationConfig()))
-          .setType("string");
-      for (Object enumConstant : type.getRawType().getEnumConstants()) {
-        builder.addEnumValue(enumConstant.toString());
-        builder.addEnumDescription("");
+              .setName(Types.getSimpleName(type, config.getSerializationConfig()))
```
Comment (tangiel) [inline endpoints-framework/src/main/java/com/google/api/server/spi/config/model/SchemaRepository.java]: Is there a reason for changing this indent?

File: test-utils/src/main/java/com/google/api/server/spi/testing/TestEnumDescription.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (tangiel) [inline test-utils/src/main/java/com/google/api/server/spi/testing/TestEnumDescription.java]: Put 2018 here.

**Pull request merged**